### PR TITLE
Feature client-registration-activation

### PR DESCRIPTION
**Pull request by rizwans-cuelogic**
Client-Registration:
  1.signup page for client using html,css and bootstrap.

Client-Activation:
  1.Client-activation by django admin.

Test-cases:
  1. Username should be 6 character long.
  2. password should be in range of 8 to 16 character long
  3. password should contain  at least 1 uppercase letter, 1 number ,1 lowercase letter.
  4. Email  should be a valid.
  5.password and confirm password should be same.

pivotal tracker-link: https://www.pivotaltracker.com/n/projects/1970105

### CODE REVIEW

File: project1/project/ebs/models.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+def send_notification(sender,instance, *args,**kwargs):
+	"""function take single user object check whether field is_active is changed 
+		to true and if it is
+		true send email to user"""
+	try:
+		if instance.is_active != User.objects.get(id=instance.id).is_active and instance.is_active==True:
+ 			print "created is"
+ 			subject = 'Active account'
+ 			mesagge = '%s Your account has been activated now. You can login and add blogs. Click here to login.' %(instance.username)
```
Comment (ananthu-nair707) [inline project1/project/ebs/models.py]: 80 chars fix, spelling correction for message variable.

File: project1/project/ebs/static/ebs/bootstrap-3.3.7-dist/js/npm.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+require('../../js/button.js')
+require('../../js/carousel.js')
+require('../../js/collapse.js')
+require('../../js/dropdown.js')
+require('../../js/modal.js')
+require('../../js/tooltip.js')
+require('../../js/popover.js')
+require('../../js/scrollspy.js')
+require('../../js/tab.js')
+require('../../js/affix.js')
```
Comment (ananthu-nair707) [inline project1/project/ebs/static/ebs/bootstrap-3.3.7-dist/js/npm.js]: End of file fix.

File: project1/project/ebs/static/ebs/css/home.css~ [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+	padding-top:5px;
+	margin-left:30px;
+}
+.color1{
+	background-color:#9370DB;
+}
+nav{
+	background-color:#9370DB;
+}
+
```
Comment (ananthu-nair707) [inline project1/project/ebs/static/ebs/css/home.css~]: Remove extra lines. Same change everywhere applicable.

File: project1/project/ebs/models.py [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+	user=models.OneToOneField(User)
+	orgname=models.CharField(max_length=125)
+	description=models.TextField(blank=True,null=True)
+	orglogo=models.ImageField(upload_to='Companylogo/')
+	def __unicode__(self):
+		return 'Organisation :' + self.orgname
+
+def send_notification(sender,instance, *args,**kwargs):
+	"""function take single user object check whether field is_active is changed 
+		to true and if it is
```
Comment (ananthu-nair707) [inline project1/project/ebs/models.py]: proper alignment of comments.

File: project1/project/ebs/models.py [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+		true send email to user"""
+	try:
+		if instance.is_active != User.objects.get(id=instance.id).is_active and instance.is_active==True:
+ 			print "created is"
+ 			subject = 'Active account'
+ 			mesagge = '%s Your account has been activated now. You can login and add blogs. Click here to login.' %(instance.username)
+ 			from_email = settings.EMAIL_HOST_USER
+ 			send_mail(subject, mesagge, from_email, 
+ 						[instance.email, "rizwan.shaikh@cuelogic.com"],
```
Comment (ananthu-nair707) [inline project1/project/ebs/models.py]: Please avoid hardcoded mail IDs here.

**Review by ananthu-nair707 — CHANGES_REQUESTED** [review]
Please make necessary changes. Do we need all the images that you have included in the PR? Are these static images or are they uploaded/tested by user?

**Comment by rizwans-cuelogic** [comment]
yes that images are  uploaded by user(client)

File: project4/project/project/settings.py [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+	os.path.join(BASE_DIR,"bootstrap-3.3.7-dist","js","static"),
+)
+MEDIA_URL='/media_cdn/'
+MEDIA_ROOT=os.path.join(BASE_DIR,"media_cdn")
+
+EMAIL_BACKEND = 'django.core.mail.backends.smtp.EmailBackend'
+EMAIL_USE_TLS = True
+EMAIL_HOST = 'smtp.gmail.com'
+EMAIL_PORT = 587
+EMAIL_HOST_USER = 'rizwanshaikh2429@gmail.com'
```
Comment (deepalib-cuelogic) [inline project4/project/project/settings.py]: use environment variables for EMAIL_HOST_USER, EMAIL_HOST_PASSWORD

File: project4/project/ebs/views.py [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+
+def registration(request):
+	return render(request,'ebs/Registration.html')
+
+
+def regresult(request):
+	"""Views for registering client to plateform take UserForm and OrgForm save it to User object
+	and organisation object"""
+	if request.method=='POST':
+		f1=UserForm(request.POST or None,request.FILES or None)
```
Comment (deepalib-cuelogic) [inline project4/project/ebs/views.py]: use proper variable names.Name should describe the variable purpose.

File: project4/project/ebs/static/ebs/js/registration.js [hunk 8]
```diff
@@ -0,0 +1,184 @@
...
+
+      $('body').on('hidden.bs.modal', '.modal', function () {
+        $('#myform')[0].reset();
+	     $('#myform').bootstrapValidator('resetForm',true);
+      });
+});
+
+
+/*
+    $.ajax({
```
Comment (deepalib-cuelogic) [inline project4/project/ebs/static/ebs/js/registration.js]: remove commented code if not required

File: project4/project/ebs/views.py [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+		f2=OrgForm(request.POST or None,request.FILES or None )
+		if f1.is_valid() and f2.is_valid():
+		 	user = User.objects.create_user(f1.cleaned_data['username'], f1.cleaned_data['email'], f1.cleaned_data['password'],is_active='False')
+		 	organisation=f2.save(commit=False)
+		 	organisation.user=user
+		 	"""g = Group.objects.get(name='client')
```
Comment (deepalib-cuelogic) [inline project4/project/ebs/views.py]: remove commented code if not required

File: project4/project/ebs/views.py [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+from django.http import HttpResponse
+from .models import Organisation
+from .forms import UserForm,OrgForm
+from django.contrib.auth.models import User
+from django.forms import ModelForm
+from django.contrib import messages
+from django.contrib.auth.models import Group
+from django import forms
+import sys
+# Create your views here.
```
Comment (deepalib-cuelogic) [inline project4/project/ebs/views.py]: Imports should be grouped in the following order:
standard library imports
related third party imports
local application/library specific imports
You should put a blank line between each group of imports.

File: project4/project/ebs/views.py [hunk 11]
```diff
@@ -0,0 +1,64 @@
...
+	return render(request,'ebs/forgotpass.html')
+
+def resetpassword(request):
+	return render(request,'ebs/newpassword.html')
+
+def registration(request):
+	return render(request,'ebs/Registration.html')
+
+
+def regresult(request):
```
Comment (deepalib-cuelogic) [inline project4/project/ebs/views.py]: can we have better name than regresult? Functions should have descriptive meaningful names that relate to their purpose.

File: project4/project/ebs/views.py [hunk 12]
```diff
@@ -0,0 +1,64 @@
...
+	return render(request,'ebs/adminlogin.html')
+
+def forgotpass(request):
+	return render(request,'ebs/forgotpass.html')
+
+def resetpassword(request):
+	return render(request,'ebs/newpassword.html')
+
+def registration(request):
+	return render(request,'ebs/Registration.html')
```
Comment (deepalib-cuelogic) [inline project4/project/ebs/views.py]: keep these things consistent.rename template to registration.html

File: project4/project/ebs/views.py [hunk 13]
```diff
@@ -0,0 +1,64 @@
...
+
+def home(request):
+	f1=UserForm();
+	f2=OrgForm();
+	return render(request,'ebs/home.html',{'f1':f1,'f2':f2})
+
+def adminlogin(request):
+	return render(request,'ebs/adminlogin.html')
+
+def forgotpass(request):
```
Comment (deepalib-cuelogic) [inline project4/project/ebs/views.py]: Use the function naming rules: lowercase with words separated by underscores as necessary to improve readability.

File: project4/project/ebs/forms.py [hunk 14]
```diff
@@ -0,0 +1,55 @@
+from django.forms import ModelForm
+from ebs.models import Organisation
+from django.contrib.auth.models import User
+from django import forms
+
```
Comment (deepalib-cuelogic) [inline project4/project/ebs/forms.py]: Arrange all imports grouped as per package and alphabetically

File: project4/project/ebs/models.py [hunk 15]
```diff
@@ -0,0 +1,40 @@
+from __future__ import unicode_literals
+from django.contrib.auth.models import User 
+from django.db import models
+from django.db.models.signals import pre_save, post_save
+from django.db.models import signals
+from django.dispatch import receiver
+from django.conf import settings
+from django.core.mail import send_mail
+
```
Comment (deepalib-cuelogic) [inline project4/project/ebs/models.py]: Arrange all imports grouped as per package and alphabetically

File: project4/project/ebs/models.py [hunk 16]
```diff
@@ -0,0 +1,40 @@
...
+ 			subject = 'Welcome To News Magzine'
+			message = 'Your account has been activated now. You can login and add blogs. click here to login ://http:'+settings.HOST+'/ebs/home'
+ 			from_email = settings.EMAIL_HOST_USER
+ 			send_mail(subject, message, from_email, 
+ 						[instance.email], 
+ 						fail_silently=True)
+ 		else:
+ 			pass
+ 	except:
+ 		print "caught"
```
Comment (deepalib-cuelogic) [inline project4/project/ebs/models.py]: add proper error message or print log the actual exception.refer https://docs.python.org/3/tutorial/errors.html

File: project4/project/ebs/views.py [hunk 17]
```diff
@@ -0,0 +1,64 @@
...
+			if 'username' in errorstring and errorstring['username']==[u'username already exists']:
+				response = {'status':'Error', 'message': "username already exists"}
+			elif 'email' in errorstring and errorstring['email']==[u'Email already exists']:	
+		 		response = {'status':'Error', 'message': "email already exists"}	
+			else:
+				response = {'status':'Error', 'message': "please fill the details"}
+			
+			return HttpResponse(json.dumps(response), content_type='application/json')
+	else:
+	 	f1=UserForm();
```
Comment (deepalib-cuelogic) [inline project4/project/ebs/views.py]: Always surround these binary operators with a single space on either side

File: project4/project/ebs/views.py [hunk 18]
```diff
@@ -0,0 +1,64 @@
...
+
+def resetpassword(request):
+	return render(request,'ebs/newpassword.html')
+
+def registration(request):
+	return render(request,'ebs/Registration.html')
+
+
+def regresult(request):
+	"""Views for registering client to plateform take UserForm and OrgForm save it to User object
```
Comment (deepalib-cuelogic) [inline project4/project/ebs/views.py]: avoid spelling mistakes

File: project4/project/ebs/templates/ebs/Registration.html [hunk 19]
```diff
@@ -0,0 +1,244 @@
...
+<head>
+	<meta charset="utf-8">	
+	<title>Registration</title>
+	<meta http-equiv="X-UA-Compatible" content="IE=edge">
+	<meta name="viewport" content="width=device-width,initial-scale=1">
+	<link rel="stylesheet" type="text/css" href="{% static 'ebs/bootstrap-3.3.7-dist/css/bootstrap.min.css' %}"/>
+	<link rel="stylesheet" type="text/css" href="{% static 'ebs/bootstrap-3.3.7-dist/css/bootstrap-theme.min.css' %}"/>
+	<link rel="stylesheet" type="text/css" href="{% static 'ebs/css/register.css' %}"/>
+	</head>
+	<style>
```
Comment (deepalib-cuelogic) [inline project4/project/ebs/templates/ebs/Registration.html]: add this in separate css file and include accordingly

File: project4/project/ebs/static/ebs/css/home.css~ [hunk 20]
```diff
@@ -0,0 +1,320 @@
+.margin1{
```
Comment (deepalib-cuelogic) [inline project4/project/ebs/static/ebs/css/home.css~]: remove this temporary css file

File: project4/project/ebs/models.py [hunk 21]
```diff
@@ -0,0 +1,40 @@
...
+
+	user=models.OneToOneField(User)
+	orgname=models.CharField(max_length=125)
+	description=models.TextField(blank=True,null=True)
+	orglogo=models.ImageField(upload_to='Companylogo/')
+	def __unicode__(self):
+		return 'Organisation :' + self.orgname
+
+def send_notification(sender,instance, *args,**kwargs):
+	"""function take single user object check whether field is_active is changed
```
Comment (deepalib-cuelogic) [inline project4/project/ebs/models.py:26]: Limit all lines to a maximum of 80 characters.

File: project4/project/ebs/tests.py [hunk 22]
```diff
@@ -0,0 +1,37 @@
+from django.test import TestCase
+import unittest
+import os
+from .models import Organisation
+from .forms import UserForm
+from django.core.mail import send_mail
+import re
+from django.contrib.auth.models import User
+from django.contrib.auth.models import User
```
Comment (deepalib-cuelogic) [inline project4/project/ebs/tests.py]: remove unnecessary imports

File: project4/project/ebs/tests.py [hunk 23]
```diff
@@ -0,0 +1,37 @@
...
+		form = UserForm(data=data)
+		self.assertTrue(form.is_valid())
+
+	def test_invalid_form(self):
+		w =User.objects.create(username='Foo', password='')
+		data = {'username': w.username, 'password': w.password,}
+		form = UserForm(data=data)
+		self.assertFalse(form.is_valid())
+
+
```
Comment (deepalib-cuelogic) [inline project4/project/ebs/tests.py:33]: need valid test cases on view functions

**Review by ananthu-nair707 — APPROVED** [review]

**Pull request merged**